expression/integration_test.go

```diff
@@ -3169,6 +3172,66 @@ func (s *testIntegrationSuite) TestIssues(c *C) {
 	tk.MustQuery("select * from t where cast(a as binary)").Check(testkit.Rows("1"))
 }
 
+func (s *testIntegrationSuite) TestFilterExtractFromDNF(c *C) {
+	tk := testkit.NewTestKit(c, s.store)
+	defer s.cleanEnv(c)
+	tk.MustExec("use test")
+	tk.MustExec("drop table if exists t")
+	tk.MustExec("create table t(a int, b int, c int)")
+
+	tests := []struct {
+		exprStr   string
+		result    string
+		remained  string
+		resultStr string
+	}{
+		{
+			exprStr: "a = 1 or a = 1 or a = 1",
+			result:  "[eq(test.t.a, 1)]",
+		},
+		{
+			exprStr: "a = 1 or a = 1 or (a = 1 and b = 1)",
+			result:  "[eq(test.t.a, 1) eq(test.t.b, 1)]",
```

Comment:
This is not correct? It should be `a = 1`, not `a = 1 and b = 1`.

Reply:
No, the extracted part is `a = 1`, and the remained part is `b = 1`. So the result is `a = 1 and b = 1`

Reply:
Consider the assignement that `a = 1, b = 2`, it is true for `a = 1 or a = 1 or (a = 1 and b = 1)`, but false for `a = 1 and b =1`.

Reply:
Now it's fixed.